### ISSUE TITLE
`tf.nn.weighted_moments` outputs NaN when `axes=[]` and `x` is complex

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46995]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A
-

**Describe the current behavior**
`tf.nn.weighted_moments` outputs NaN when `axes=[]` and `x` is complex
**Describe the expected behavior**
expect no Nan in output

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
import numpy as np
tf.nn.weighted_moments(x=np.array([1e+38+0.j, 2e+38+0.j], dtype=np.complex64), frequency_weights=10, axes=[])
~~~

Output:
~~~python
(<tf.Tensor: shape=(2,), dtype=complex64, numpy=array([inf+nanj, inf+nanj], dtype=complex64)>, <tf.Tensor: shape=(2,), dtype=complex64, numpy=array([inf+nanj, inf+nanj], dtype=complex64)>)
~~~

**Comment by ravikyram** [comment]
Was able to reproduce the issue with TF v2.1, TF v2.4 and TF-nightly(`2.5.0-dev20210204`). Please find the gist of it [here](https://colab.research.google.com/gist/ravikyram/178106455baefc09e20739cdaded9c97/untitled663.ipynb). Thanks!

**Comment by saikumarchalla** [comment]
Issue still exists in TF 2.6 and please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/8b60e3cd2d1c73fb3c106b1b598572e5/untitled92.ipynb).Thanks!

**Comment by SeeForTwo** [comment]
Thank you for filing this issue.
Large _x_ values multiplied by large _frequency_weights_ with a result greater than about 3.4e+38 causes overflow in float32 representation (which is used by np.complex64). As a work around for cases where avoiding this kind of overflow is appropriate, the caller can normalize frequency_weights to have small values (e.g. by using a floating point type and dividing by the maximum value so that the maximum value of the normalized result is 1.0). However, there is no generic way to rescale weights that would not conflict with a valid use case (e.g. that would not underflow for small values). Therefore, we decided to not change the behavior.

**Comment by tilakrayal** [comment]
@DNXie,
As the work-around mentioned by the developer, I tried with the value less than `3.4e+38` i.e., **[3.4e+37+0.j, 3.4e+37+0.j]** and it was executed with the expected output. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/7194d13b8569d3a643826f76e70628d3/untitled1562.ipynb) and also please try to normalize `frequency_weights` to have small values for avoiding this kind of overflow. Thank you!
